src/mono/mono/mini/mini.h

```diff
@@ -717,15 +717,15 @@ typedef struct {
 	LLVMArgInfo args [1];
 } LLVMCallInfo;
 
-#define MONO_MAX_SRC_REGS	3
+#define MONO_MAX_SRC_REGS	4
 
 struct MonoInst {
  	guint16 opcode;
 	guint8  type; /* stack type */
 	guint8  flags;
 	
 	/* used by the register allocator */
-	gint32 dreg, sreg1, sreg2, sreg3;
+	gint32 dreg, sreg1, sreg2, sreg3, sreg4;
```

Comment:
There was a fair bit of discussion about this when I looked into it, and I thought the conclusion was that we didn't want to add them this way since it's only an immediate value.

Reply:
This adds a lot of overhead for the tiny minority of opcodes which have 4 opcodes.

Reply:
There's 4 bytes of padding between `sreg3` and `next` on 64-bit machines.

Reply:
That only appears to be true for Clang/GCC on x64; other architectures and MSVC do not appear to match that behavior. We also still run on a (sadly) 32-bit platform – wasm – so I don't think we can assume this doesn't add overhead on platforms we care about.